client/src/includes/panels.ts

```diff
@@ -105,12 +107,18 @@ export function initCollapsiblePanels(
 /**
  * Smooth scroll onto any active panel.
  * Needs to run after the whole page is loaded so the browser can resolve any
- * JS-rendered :target.
+ * JS-rendered elements.
  */
 export function initAnchoredPanels(
-  anchorTarget = document.querySelector<HTMLElement>('[data-panel]:target'),
+  anchorTarget = document.getElementById(window.location.hash.slice(1)),
 ) {
-  if (anchorTarget) {
-    anchorTarget.scrollIntoView({ behavior: 'smooth' });
+  const target = anchorTarget?.matches('[data-panel]')
+    ? anchorTarget
+    : getElementByContentPath();
+
+  if (target) {
+    setTimeout(() => {
+      target.scrollIntoView({ behavior: 'smooth' });
+    }, 100);
```

Comment:
Feel free to experiment. I tried using no timeout at all and using a smaller value, but they didn't work. It seems `100` works consistently enough.

Try the following scenarios:

- Using the built-in "link" anchor feature
- Using the `contentpath` directive. Find an element with `data-contentpath` that you want to scroll to, which may or may not have the built-in anchor feature e.g. in `MultiFieldPanel`s and `StructBlock`s. Examples:
  - Block inside `StructBlock`, no built-in anchor: http://localhost:8000/admin/pages/82/edit/#:w:contentpath=e0b41ad1-cae5-482d-ae31-f13b35f17830.difficulty 
  - `StructBlock` itself, has built-in anchor: http://localhost:8000/admin/pages/82/edit/#:w:contentpath=e0b41ad1-cae5-482d-ae31-f13b35f17830
  - `FieldPanel` inside `MultiFieldPanel`, no built-in anchor: http://localhost:8000/admin/pages/82/edit/#:w:contentpath=recipe_headline
  - Top-level `FieldPanel`, has built-in anchor: http://localhost:8000/admin/pages/82/edit/#:w:contentpath=subtitle

Note that in some cases when using the `contentpath` directive, the browser may scroll directly to the input field and the label isn't visible. This is most noticeable in the top-levle `FieldPanel` case. This is because for some cases the label is within the element with `data-contentpath`, while in other cases it is outside. We could in theory try to find the label by traversing the ancestors/descendants and scroll to that element instead, but if we're not careful we may end up scrolling to the "parent" content instead of the most specific node.


Also, I recommend testing with recipe pages on bakery demo _but_ with the panels set on the Promote tab instead of the content tab, to ensure this all still works when the target element is in a tab that's not the first one. You can do this easily by changing `content_panels = Page.content_panels + [` to `promote_panels = Page.promote_panels + [`.